### DISCUSSION TITLE
Remove Segment From V2 Message

### DISCUSSION THREAD

**Discussion by LLCJK** [nextgenhealthcare/connect#5383] [Q&A] [answered]
I am trying to create a channel that grabs values from a note segment and then removes the segments completely from the original message.

var count = msg['NTE'].length()
var builtString = ""

for(x=0; x<count; x++){
	builtString = builtString + ' ' + msg['NTE'][x]['NTE.3']['NTE.3.1'].toString()
	delete msg['NTE'][x]
}

I would think that this would remove the whole NTE[x] segment but I am instead getting a "DETAILS:	TypeError: Cannot read property "NTE.3" from undefined" error.

**Comment by pacmano1** [comment, marked as answer]
remember when you delete a segment you are "rearranging" the NTE array.

See https://forums.mirthproject.io/forum/mirth-connect/support/181472-combine-segments-and-delete-issue (ignore the combining stuff, I think you are having the same problem).

**Reply by pacmano1** [reply]
Well, I think that link I posted is exactly your use case, lol.

**Reply by LLCJK** [reply]
Once you mentioned the "rearranging" I recognized exactly what I needed to do. You have to start from the last NTE and work your way backwards. Only issue with this now is that the builtString also builds backwards which isn't exactly what I want..

for(ix=count-1; x>=0; x--){
	builtString = builtString + ' ' + msg['NTE'][x]['NTE.3']['NTE.3.1'].toString()
	delete msg['NTE'][x]
}

**Reply by pacmano1** [reply]
I covered that also in that post.

**Reply by LLCJK** [reply]
I see that I am going to need the if statement, but what would 'SAMPLE' be in my case?

**Reply by pacmano1** [reply]
```JS
var count = msg['NTE'].length()
var builtString = ''
for (x = count - 1; x >= 0; x--) {
    builtString = msg['NTE'][x]['NTE.3']['NTE.3.1'].toString() + ' ' + builtString
    delete msg['NTE'][x]
}
//  but sort that space out, not a computer
```

And please use code blocks and formatting when you post here.
